roles/openshift_efk/tasks/main.yaml

```diff
@@ -0,0 +1,179 @@
+    register: secret_output
+    failed_when: "secret_output.rc == 1 and 'exists' not in secret_output.stderr"
+    tags:
+      - build
+
+  - name: "Copy serviceAccount file"
+    copy: dest=/tmp/logging-deployer-sa.yaml
+          src={{role_path}}/files/logging-deployer-sa.yaml
+          force=yes
+    tags:
+      - build
+
+  - name: "Create logging-deployer service account"
+    shell: oc create -f  /tmp/logging-deployer-sa.yaml
+    register: deployer_output
+    failed_when: "deployer_output.rc == 1 and 'exists' not in deployer_output.stderr"
+    tags:
+      - build
+
+  - name: "Set permissions for logging-deployer service account"
+    command: oc policy add-role-to-user edit system:serviceaccount:logging:logging-deployer
+    register: permiss_output
+    failed_when: "permiss_output.rc == 1 and 'exists' not in permiss_output.stderr"
+    tags:
+      - build
+
+  - name: "Set permissions for fluentd"
+    command: oadm policy add-scc-to-user privileged system:serviceaccount:logging:aggregated-logging-fluentd
+    register: fluentd_output
+    failed_when: "fluentd_output.rc == 1 and 'exists' not in fluentd_output.stderr"
+    tags:
+      - build
+
+  - name: "Set additional permissions for fluentd"
+    command: oadm policy add-cluster-role-to-user cluster-reader system:serviceaccount:logging:aggregated-logging-fluentd
+    register: fluentd2_output
+    failed_when: "fluentd2_output.rc == 1 and 'exists' not in fluentd2_output.stderr"
+    tags:
+      - build
+
+  - name: "Make sure to remove stale deployer template"
+    command: oc delete template logging-deployer-template -n openshift
+    register: delete_ouput
+    failed_when: delete_ouput.rc == 1 and 'exists' not in delete_ouput.stderr
+    tags:
+      - build
+
+  - name: "Create deployer template"
+    command: oc create -f /usr/share/openshift/examples/infrastructure-templates/enterprise/logging-deployer.yaml -n openshift
+    register: template_output
+    failed_when: "template_output.rc == 1 and 'exists' not in template_output.stderr"
+    tags:
+      - build
+
+  - name: "Clear out any previous pods"
+    command: oc delete pods --all
+    tags:
+      - build
+
+  - name: "Process the deployer template with an registry other than registry.access.redhat.com"
+    shell: oc process logging-deployer-template -n openshift -v KIBANA_HOSTNAME={{ kibana_hostname | quote }},ES_CLUSTER_SIZE={{ es_cluster_size | quote }},PUBLIC_MASTER_URL={{ master_url | quote }},IMAGE_PREFIX={{ target_registry | quote }}/  | oc create -f -
+    when: target_registry is defined
+    tags:
+      - build
+
+  - name: "Process the default deployer template"
+    shell: oc process logging-deployer-template -n openshift -v KIBANA_HOSTNAME={{ kibana_hostname | quote }},ES_CLUSTER_SIZE={{ es_cluster_size | quote }},PUBLIC_MASTER_URL={{ master_url | quote }}  | oc create -f -
+    when: target_registry is not defined
+    tags:
+      - build
```

Comment:
You can consolidate the two process deployer template tasks as follows:
```
- name: Process the deployer template with an registry other than registry.access.redhat.com
  shell: >
    oc process logging-deployer-template -n openshift -v
    KIBANA_HOSTNAME={{ kibana_hostname | quote }},ES_CLUSTER_SIZE={{ es_cluster_size | quote }},PUBLIC_MASTER_URL={{ master_url | quote }},{{ 'IMAGE_PREFIX=' ~ target_registry | quote if target_registry is defined else '' }}
    | oc create -f -
```

It can be cleaned up a bit further:
```
- name: Process the deployer template with an registry other than registry.access.redhat.com
  shell: oc process logging-deployer-template -n openshift -v {{ oc_process_values }} | oc create -f -
```
in vars/main.yml:
```
kh_kv: KIBANA_HOSTNAME={{ kibana_hostname | quote }}
es_cs_kv: ES_CLUSTER_SIZE={{ es_cluster_size | quote }}
pmu_kv: PUBLIC_MASTER_URL={{ master_url | quote }}
ip_kv: "{{ 'IMAGE_PREFIX=' ~ target_registry | quote if target_registry is defined else '' }}"
oc_process_values: "{{ kh_kv }} {{ es_cs_kv }} {{ pmu_kv }} {{ ip_kv }}"
```